corehq/util/couch_doc_processor.py

```diff
@@ -212,7 +212,7 @@ def filter(self, doc):
         return True
 
 
-def process_docs(slug, doc_type_map, doc_migrator, reset=False, max_retry=2, chunk_size=100):
+class CouchDocumentProcessor(object):
```

Comment:
The point of doing this was so that you could get access to the `ResumableDocsByTypeIterator` to get the progress. See `has_started` method